browser/ai_chat/tab_tracker_service_factory.cc

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) 2025 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#include "brave/browser/ai_chat/tab_tracker_service_factory.h"
+
+#include <memory>
+
+#include "brave/browser/ai_chat/ai_chat_service_factory.h"
+#include "brave/browser/ai_chat/ai_chat_utils.h"
+#include "brave/components/ai_chat/core/browser/tab_tracker_service.h"
+#include "components/keyed_service/content/browser_context_dependency_manager.h"
+
+namespace ai_chat {
+
+// static
+TabTrackerServiceFactory* TabTrackerServiceFactory::GetInstance() {
+  static base::NoDestructor<TabTrackerServiceFactory> instance;
+  return instance.get();
+}
+
+// static
+TabTrackerService* TabTrackerServiceFactory::GetForBrowserContext(
+    content::BrowserContext* context) {
+  CHECK(context);
+
+  // If AIChat isn't allowed for this context, we don't need a
+  // TabTrackerService.
+  if (!IsAllowedForContext(context)) {
```

Comment:
This is the wrong place for this because it will still get created for contexts that don't use it in `ServiceIsCreatedWithBrowserContext`
The proper place for this is `GetBrowserContextToUse`